components/tools/OmeroPy/test/integration/gatewaytest/test_wrapper.py

```diff
@@ -156,6 +158,58 @@ def testDetailsWrapper(self, gatewaywrapper, author_testimg):
         assert d.getGroup().name == \
             img.getProject().getDetails().getGroup().name
 
+    def createTestFile(self, gatewaywrapper):
+        gatewaywrapper.loginAsAuthor()
+        content = 'abcdefghijklmnopqrstuvwxyz'
+        f = gatewaywrapper.createTestFile(
+            'testOriginalFileWrapper', 'test_wrapper', content)
+        return f
+
+    def testOriginalFileWrapperGetFileInChunks(self, gatewaywrapper):
+        f = self.createTestFile(gatewaywrapper)
+        assert ''.join(f.getFileInChunks()) == 'abcdefghijklmnopqrstuvwxyz'
+        assert list(f.getFileInChunks(buf=10)) == [
+            'abcdefghij',
+            'klmnopqrst',
+            'uvwxyz',
+        ]
+
+    def testOriginalFileWrapperAsFileObj(self, gatewaywrapper):
+        f = self.createTestFile(gatewaywrapper)
+        fobj = f.asFileObj(buf=7)
+        assert fobj.pos == 0
+        assert fobj.read(5) == 'abcde'
+        assert fobj.pos == 5
+        fobj.seek(10)
+        assert fobj.pos == 10
+        assert fobj.read(5) == 'klmno'
+        assert fobj.pos == 15
+        fobj.seek(5, os.SEEK_CUR)
+        assert fobj.pos == 20
+        assert fobj.read(4) == 'uvwx'
+        fobj.seek(-5, os.SEEK_END)
+        assert fobj.pos == 21
+        assert fobj.read(2) == 'vw'
+        assert fobj.read() == 'xyz'
+        assert fobj.read() == ''
+        fobj.close()
+
+    def testOriginalFileWrapperAsFileObjContextManager(self, gatewaywrapper):
+        f = self.createTestFile(gatewaywrapper)
+        with f.asFileObj() as fobj:
+            assert fobj.read() == 'abcdefghijklmnopqrstuvwxyz'
+        # Verify close was automatically called
+        with pytest.raises(Ice.ObjectNotExistException):
+            fobj.read()
+
+    def testOriginalFileWrapperAsFileObjMultiple(self, gatewaywrapper):
+        # Test that multiple file objects are allowed
+        # https://trello.com/c/lC8hFFix/522
+        f1 = self.createTestFile(gatewaywrapper)
+        f2 = self.createTestFile(gatewaywrapper)
+        with f1.asFileObj() as fobj1, f2.asFileObj() as fobj2:
```

Comment:
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-integration-python/990/testReport/junit/(root)/(empty)/OmeroPy_test_integration_gatewaytest_test_wrapper/ shows a problem: perhaps a Python-2.7-ism?

Reply:
Thanks, should be fixed now